FirebaseMessaging/Sources/FIRMessaging.m

```diff
@@ -239,11 +239,13 @@ - (void)didCompleteConfigure {
                                                        scope:kFIRMessagingDefaultTokenScope]
           .token;
   // When there is a cached token, do the token refresh.
+  // Before fetching FCM token, confirm there is an APNS token to avoid validation error
+  // This error is innocuous since we refetch after APNS token is set but it can seem alarming
   if (cachedToken) {
     // Clean up expired tokens by checking the token refresh policy.
     [self.installations installationIDWithCompletion:^(NSString *_Nullable identifier,
                                                        NSError *_Nullable error) {
-      if ([self.tokenManager checkTokenRefreshPolicyWithIID:identifier]) {
+      if ([self.tokenManager checkTokenRefreshPolicyWithIID:identifier] && self.APNSToken) {
```

Comment:
I wonder if we should guard the check in a lower level where always check APNS token is valid before sending a get FCM token request

Reply:
Thats a good point - the lower check already exists in the TokenManager. However, that returns an Error. The error is not harmful and we refetch once APNS token is set but the Error causes an alarm for developers and I think triggers Error reports. 

This check is so we don't begin fetch it in the configure flow itself - slightly higher level than the TokenManager. This will prevent false errors.

Reply:
Also, we already have a flow that refetches once an APNS token is set. This is to prevent a default configure flow we have.